csrc/id_model/indexing.cpp

```diff
@@ -59,12 +59,23 @@ bool isPartitionedLoop(TensorView* tv, IterDomain* id) {
 }
 
 bool isSizeOneDomain(IterDomain* id) {
-  return id->isBroadcast() || id->isReduction() || id->extent()->isOneInt();
+  return id->isBroadcast() || id->extent()->isOneInt();
 }
 
-// True if a given domain of a tensor *may* require allocation
-bool mayRequireAllocation(TensorView* tv, IterDomain* id) {
-  return !isPartitionedLoop(tv, id) && !isSizeOneDomain(id);
+// True if a given domain of a tensor *may* require allocation. The
+// optional promotion domain is also considered if given.
+bool mayRequireAllocation(
+    TensorView* tv,
+    IterDomain* id,
+    IterDomain* promotion_id = nullptr) {
+  // Conditions to consider:
+  // - Fully partitioned: Check the original ID, not the promotion ID
+  // - Size one: Check the promotion ID if given, the original ID otherwise.
```

Comment:
If `promotion_id` is not size 1 but `id` is size 1, why do we still need to allocate that dimension for `id`?

Reply:
That's a good point. As long as the original domain is size one, I believe we can keep it as is, so its allocation should not need to be expanded. Let me test the change.

Reply:
Cleaned it up. Testing the [tracking branch](#2238 ) also passed.